xbmc/cores/VideoPlayer/DVDSubtitles/DVDSubtitlesLibass.h

```diff
@@ -26,8 +37,29 @@ class CDVDSubtitlesLibass
   */
   void Configure();
 
-  ASS_Image* RenderImage(int frameWidth, int frameHeight, int videoWidth, int videoHeight, int sourceWidth, int sourceHeight,
-                         double pts, int useMargin = 0, double position = 0.0, int* changes = NULL);
+  ASS_Image* RenderImage(double pts,
+                         KODI::SUBTITLES::LibassRenderOpts opts,
+                         bool updateStyle,
+                         KODI::SUBTITLES::subtitlesStyle subStyle,
+                         int* changes = NULL);
+
+  /*!
+  * \brief Specifies whether the style has been initialised
+  */
+  bool IsStyleInitialized();
+
+  /*!
+  * \brief Translate and apply Kodi subtitles style to LibAss style
+  */
+  void ApplyStyle(KODI::SUBTITLES::subtitlesStyle subStyle);
+
+  /*!
+  * \brief Specify whether the subtitles are
+  * native (loaded from ASS/SSA file or stream)
+  * or adapted (converted from other types e.g. SubRip)
+  */
+  void SetSubtitleType(ASSSubType type) { m_subtitleType = type; };
```

Comment:
Doesn't need to be public afaics. I thing you don't even need the getter as this is restricted to the class instance

Reply:
this is used by subtitlesAdapter

Reply:
i could add in the libass Configure() method an argument to be used from subtitlesAdapter to set it instead of SetSubtitleType

Reply:
You're right. Leave it as is as it's easier to understand. We will need to revisit this before approval anyway. I still think that all the methods that are called from the adapter should be private or protected in libass so that we know no other components are supposed to call them. This can be done either by making the "adapter" extend the libass handler or setting the adapter as a friend class of the handler (See https://www.google.com/amp/s/www.geeksforgeeks.org/friend-class-function-cpp/amp/). I'll look into it later

Reply:
i have already think this before you, if you extend the subtitlesAdapter class to libass class, then you cannot do libass as std::make_shared this means having to handle different points of the overlayRenderer and overlays and parsers in more complex way,
and also you have to redo all the PR code from the beginning
i think that has no sense
the current implementation is the most user-friendly approach, also for understanding the functionality from external users

Reply:
if you want do this i stop develop the PR now because it is a lost of time without sense

Reply:
if the problem is only this SetSubtitleType i move it to the Configure method as argument
it is not a problem, but redoing everything from scratch makes no sense and I'm not willing to do that

Reply:
I clearly misinterpreted the use of "friend class", please try look at last pushed commit

Reply:
Yeah I  didn't understand your reaction as the suggested change was quite small, literally setting a class as a friend and move stuff to private or protected :) I thought you could understand what I meant from the shared link.

It might depend a bit on the interpretation though. Abusing friend classes is a bad practice, in this case I think it is better to hide the methods that shouldn't be publicly used by anything rather than our "adapter"/proxy than simply having all of them public.